mod_cdrs.asciidoc

```diff
@@ -301,19 +302,19 @@ Example: A driver selects a different <<mod_sessions.asciidoc#mod_sessions_set_c
 <<mod_sessions.asciidoc#mod_sessions_profile_type_enum,`profile_type`>> during an ongoing charging session, the different profile might have a different tariff.
 
 The `step_size` uses the total amount of a certain unit used during a session, not only the last ChargingPeriod.
-In other words, when charging tariff per kWh of time differs during a session, the total amount of kWh of time is used in calculations with `step_size`. +
-Example: Charging cost 0.20 euro/Wh before 17:00 and 0.27 euro/Wh after 17:00 both have a `step_size` of 500 Wh.
+In other words, when the price of energy per kWh or the price of time per hour differs during a session,only the total amount of energy or time is used in calculations with `step_size`. +
+Example: Energy costs € 0.20 perkWh before 17:00 and € 0.27 per kWh after 17:00. Both Price Components have a `step_size` of 500 Wh.
 If a driver charges 4.3 kWh before 17:00 and 1.1 kWh after 17:00, a total of 5.4 kWh is charged. The `step_size` rounds this up to 5.5 kWh total.
 It does NOT round the energy used after 17:00 to 1.5 kWh.
 
-Example: Charging cost 5 euro/hour before 17:00 and 7 euro/hour after 17:00 both have a `step_size` of 10 minutes.
-If a driver charges 6 minutes before 17:00 and 22 minutes after 17:00: a total of 28 minutes charging. The `step_size` rounds this up to 30 minutes total, so 24 minutes after 17:00 will be billed.
+Example: Time costs € 5 per hour before 17:00 and € 7 per hour after 17:00. Both Price Components have a `step_size` of 10 minutes.
+If a driver charges 6 minutes before 17:00 and 22 minutes after 17:00, this makes a total of 28 minutes charging. The `step_size` rounds this up to 30 minutes total, so 24 minutes after 17:00 will be billed.
 It does NOT round the minutes after 17:00 to 30 minutes, which would have made a total of 36 minutes.
 
 In the cases that <<mod_tariffs.asciidoc#mod_tariffs_tariffdimensiontype_enum,`TIME`>>
 and <<mod_tariffs.asciidoc#mod_tariffs_tariffdimensiontype_enum,`PARKING_TIME`>> Tariff Elements are both used, `step_size` is only taken into account for the total parking duration` +
-Example: Charging cost 1.00 euro/hour, parking 2.00 euro/hour both have a `step_size` of 10 minutes.
-If a driver charges 21 minutes, and keeps his EV connected while it is full for another 16 minutes. The step_size rounds the parking duration up to 20 minutes, making it a total of 41 minutes. Note that the charging duration is not rounded up, as it is followed by another time base period.
+Example: Time spent charging costs € 1.00 per hour and time spent parking (not charging) costs € 2.00 per hour. Both Price Components have a `step_size` of 10 minutes.
+If a driver charges 21 minutes, and keeps his EV connected while it is full for another 16 minutes, then the step_size rounds the parking duration up to 20 minutes, making it a total of 41 minutes. Note that the charging duration is not rounded up, as it is followed by another time based period.
 
```

Comment:
> This seems pretty counter intuitive to me, imho. I'd like to see a tariff with different step size, and weird ones. What happens to a customer that start charging at 6h59 for 6 hours, when the tariff is:
> 
> from 7h to 17h: 1€ per hour, step size of 1 hour. from 17h to 7h: 0€ per hour, step size of 6 hour.
> 
> (don't laugh, it's almost a copycat of one our tariff in our system)
> 
> In this case the customer would pay nothing for 6 hours, just because he was one minute before 7h ? I think (again, in this case only, I'm pretty sure we could find counter examples) it would be better to make the client pay for the 6 whole hours.
> 
> Even in the case of a charge finishing at 7h01, I think in this case it would be better that the customer pays one hour.

I'd calculate your example like this:

- Customer starts charging at 06:59. At this point a price component becomes active with €0,- per hour with a step-size of 6 hours.
- at 07:00 this period of 1 minute ends and the price component with €1,- per hour with a step-size of 1 hour becomes active.
- at 12:59 the session end.

Now the two periods are priced as following:
- 1 minute of charging time costing a total of €0; (step size is not taken into account yet since it is followed by another time based price-component)
- 5 hours and 59 minutes of charging time are rounded up to 6 hours (step-size of 1 hour) costing a total of €6,-.

Reply:
From memory, I think what @remkop22 does is what OCPI 2.2.1 does.